src/main/java/org/elasticsearch/action/search/ClearScrollRequest.java

```diff
@@ -19,51 +19,77 @@
 
 package org.elasticsearch.action.search;
 
+import org.elasticsearch.ElasticsearchIllegalArgumentException;
 import org.elasticsearch.action.ActionRequest;
 import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.bytes.BytesReference;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
 
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.List;
 
-import static com.google.common.collect.Lists.newArrayList;
-import static org.elasticsearch.action.ValidateActions.addValidationError;
+import static org.elasticsearch.action.ValidateActions.*;
 
 /**
  */
 public class ClearScrollRequest extends ActionRequest<ClearScrollRequest> {
 
-    private List<String> scrollIds;
+    private BytesReference source;
+    private boolean unsafe;
```

Comment:
the unsafe field can be removed from ClearScrollRequest, because in TransportClearScrollAction we don't fork the request. Everything happens on the calling thread.